### DISCUSSION TITLE
[Request title]: Website to display modeling results

### DISCUSSION THREAD

**Discussion by usamabilal** [Drexel-UHC/analytics-corner#12] [Infrastructure]
### Name

Usama Bilal

### Job title

assistant professor

### Department

Urban Health Collaborative

### Request type

Website for grant

### Request description

We will be modeling the effects of polices to mitigate the effects of food prices on NCDs in a number of US cities. We want to build a website that displays the results of these models by city, so that folks can click around, select their city, a policy, and see what happens.

### Data

Data is the result of models, so it's all non private (I think)

### Notes

Add Brent Langellier <bal95@drexel.edu> to issue

**Comment by ran-codes** [comment]
Converting this to a discussion as we found that these consultation/budget-estimate conversations work better as a discussion rather than an issue.

**Reply by ran-codes** [reply]
Also tagging @blangellier13 now! @blangellier13 please comment if you got this 😄

**Reply by blangellier13** [reply]
Got it!

**Comment by ran-codes** [comment]
## Technical Costs (data storage + web hosting)

technically your site will not be a problem. Model outputs are small in size and a static site built in JS would be more than enough to provide the flexibility and visualizations required. This client-side design is also extremely affordable.... so no cost. Also, we already have the expertise at UHC to execute on this plan. 

**Architecture summary:** A static website  (Azure Static Web Apps or GitHub pages) linked to some sort of data store (Azure blob storage) in which your project models results can be stored in.

**Deployment cost estimate:** Based on the proposed architecture above your hosting and data storage would be a non-factor (<1$ per month).

**Comment by ran-codes** [comment]
## Staffing Costs

This is an recomendation and numbers may fluctuate but is a good approximation. 

We will use three variables to estimate costs: 1) years of project 2) % of a person's time 3) a person's full-time yearly salary. I recommend you provision for two roles **analytics engineer** (mix of statistics + computer science + web dev) and a **communications specialist** (will help with UI/UX of your app and dissemination). 

- **Case 1: part-time engineer year 1-5**: In this case you provision funding for a part-time engineer throughout your project life cycle. This engineer would oversee your  analyst/RA/GRA to and help them to build a well organized and effeicient data and codebase infrastructure. By year 3-5 roles around your codebase and data infrastructure are well documented, well organized and immediate machine actioanble and you core staff (GRA/analysts) will be well trained; your engineer can now focus on building the web application. The cost over the project life cycle would be.
   - Analytics Engineer (year 1-5): (5 years) * (0.1 of time) * ($80k yearly salary) = $40k
   - Communications specialist (year 3-5): (3 years) * (0.05 of time) * ($60k yearly salary) = $9k
   - **Total cost: $50k**
- **Case 2: part-time engineer year 3-5**: In this case you provision funding for a part-time engineer on to build your website.  They will come in year 3-5 and have to first grapple with your existing infrastructure get it machine actionable then build the web application; in our experience this could be messy and be time consuming. 
   - Analytics Engineer (year 3-5): (3 years) * (0.25 of time) * ($80k yearly salary) = $60k
   - Communications specialist (year 3-5): (3 years) * (0.05 of time) * ($60k yearly salary) = $9k
   - **Total cost: $70k**

I recomend case 1. Having a more experience engineer will not only make bulding web site easier when it comes but also to build effective workflows, infrastructure will have exponential benefits to overall research potential and productivity. In case case 2, you might be paying a engineer to do things that the analysts or GRA should have been taught to well in years 1-3.

**Reply by blangellier13** [reply]
Option 1 sounds good -- do we have a staff engineer that we can hire at 10%?

**Reply by ran-codes** [reply]
Right now there is me. But the goal for the UHC (I hope) is to start building technical skill amongst UHC analysits or teaching graduates students these key skills, so we can move some analyst towards engineers. (right @usamabilal ?) or enable graduate students to be competive for such jobs (which are becoming much more in demand imo).

**Reply by usamabilal** [reply]
Indeed. For now lets put Ran (as he's the example of the type of analytics engineer that would do this) and when "time" comes we can see who's the person that can will do that. So this will be 10% of Ran's time for Y1-Y5 and 5% of Comms person (old Sarah, new person we will hire, if the person doesn't quit after 24h that's it)
